flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -448,8 +452,8 @@ private byte[] writeToManifest(long checkpointId) throws IOException {
     if (writeResultsOfCurrentCkpt.isEmpty()) {
       return EMPTY_MANIFEST_DATA;
     }
-
-    WriteResult result = WriteResult.builder().addAll(writeResultsOfCurrentCkpt).build();
+    List<WriteResult> writeResults = writeResultsOfCurrentCkpt.get(checkpointId);
```

Comment:
What happens with the `WriteResult` of the previous checkpoints?

The situation which lead you this case would cause data loss. The data from the first (unfinished) checkpoint needs to be committed as well, so if some rows are updated only in that given checkpoint it is stored here as well.

We should write a test which covers this scenario